index.js

```diff
@@ -77,15 +79,29 @@ function storeOptionValue(parseOptions, option, value, result) {
   }
 }
 
-const parseArgs = (
-  argv = getMainArgs(),
+const parseArgs = ({
+  args = getMainArgs(),
   options = {}
-) => {
-  validateArray(argv, 'argv');
+} = {}) => {
+  validateArray(args, 'args');
   validateObject(options, 'options');
-  for (const key of ['withValue', 'multiples']) {
-    if (ObjectHasOwn(options, key)) {
-      validateArray(options[key], `options.${key}`);
+  for (const [option, optionConfig] of ObjectEntries(options)) {
+    validateObject(optionConfig, `options.${option}`);
+
+    if (ObjectHasOwn(optionConfig, 'type')) {
+      validateUnion(optionConfig.type, `options.${option}.type`, ['string', 'boolean']);
+    }
+
+    if (ObjectHasOwn(optionConfig, 'short')) {
+      const short = optionConfig.short;
+      validateString(short, `options.${option}.short`);
+      if (short.length !== 1) {
+        throw new Error(`options.${option}.short must be a single character got "${short}"`);
```

Comment:
Cosmetic, optional suggestion:
```suggestion
        throw new Error(`options.${option}.short must be a single character, got '${short}'`);
```

Reply:
👌 https://github.com/pkgjs/parseargs/pull/63/commits/b41fd9bee71f7e6aa6dc63560a739eec3e2ca0b0